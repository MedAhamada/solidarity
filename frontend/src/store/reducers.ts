/**
 * Combine all reducers in this file and export the combined reducers.
 */

import { AnyAction, combineReducers, Reducer } from '@reduxjs/toolkit';

import { InjectedReducersType } from 'utils/types/injector-typings';
import { connectRouter, RouterState } from 'connected-react-router';
import history from '../utils/history';
import { appReducer as app } from '../app/containers/App/reducer';
import { reducer as login } from '../app/containers/Login/slice';
/**
 * Merges the main reducer with the router state and dynamically injected reducers
 */
export function createReducer(injectedReducers: InjectedReducersType = {}) {
  // Initially we don't have any injectedReducers, so returning identity function to avoid the error
  return combineReducers({
    ...injectedReducers,
    router: connectRouter(history) as Reducer<RouterState, AnyAction>,
    app,
    login,
  });
}
