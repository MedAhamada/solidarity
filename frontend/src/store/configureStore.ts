/**
 * Create the store with dynamic reducers
 */

import {
  configureStore,
  getDefaultMiddleware,
  Middleware,
} from '@reduxjs/toolkit';
import { createInjectorsEnhancer, forceReducerReload } from 'redux-injectors';
import createSagaMiddleware from 'redux-saga';
import { routerMiddleware } from 'connected-react-router';
import { History } from 'history';
import storage from 'redux-persist/lib/storage';
import { persistStore, persistReducer, PERSIST } from 'redux-persist';

import { createReducer } from './reducers';

export function configureAppStore(history?: History, preloadedState = {}) {
  const persistConfig = {
    key: 'root',
    storage,
    whitelist: [],
  };
  const persistedReducer = persistReducer(persistConfig, createReducer());

  const reduxSagaMonitorOptions = {};
  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);
  const { run: runSaga } = sagaMiddleware;

  // Create the store with saga middleware
  const middlewares = [sagaMiddleware] as Middleware[];
  if (history) {
    middlewares.push(routerMiddleware(history));
  }

  const enhancers = [
    createInjectorsEnhancer({
      createReducer,
      runSaga,
    }),
  ];

  const store = configureStore({
    reducer: persistedReducer,
    middleware: [
      ...getDefaultMiddleware({
        serializableCheck: {
          // see https://github.com/rt2zz/redux-persist/issues/988#issuecomment-529575333 to ignore actions for serializable check
          ignoredActions: [PERSIST],
        },
      }),
      ...middlewares,
    ],
    devTools: process.env.NODE_ENV !== 'production',
    preloadedState,
    enhancers,
  });
  const persistor = persistStore(store);

  /* istanbul ignore next */
  if (process.env.NODE_ENV !== 'production' && module.hot) {
    module.hot.accept('./reducers', () => {
      import('./reducers').then(() => store.replaceReducer(persistedReducer));
    });
  }
  return { store, persistor };
}
