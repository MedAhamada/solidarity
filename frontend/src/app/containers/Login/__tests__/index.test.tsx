import * as React from 'react';
import { render } from '@testing-library/react';
import { Store } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { HelmetProvider } from 'react-helmet-async';
import history from 'utils/history';

import { configureAppStore } from 'store/configureStore';
import { Login } from '..';

const renderComponent = (store: Store) =>
  render(
    <Provider store={store}>
      <HelmetProvider>
        <Login />
      </HelmetProvider>
    </Provider>,
  );

describe('<Login />', () => {
  let store: ReturnType<typeof configureAppStore>;

  beforeEach(() => {
    store = configureAppStore(history);
  });
  it('should match the snapshot', () => {
    const component = renderComponent(store.store);
    expect(component.container.firstChild).toMatchSnapshot();
  });
});
