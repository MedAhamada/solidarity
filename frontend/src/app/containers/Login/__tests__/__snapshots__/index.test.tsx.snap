// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should match the snapshot 1`] = `
<div
  class=""
>
  <h1>
    My App
  </h1>
  <p>
    This can be anywhere in your application access=
  </p>
  <form
    action="#"
  >
    <h1>
      Login
    </h1>
    <input
      name="email"
      type="email"
      value=""
    />
    <input
      name="password"
      type="password"
      value=""
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
  <button>
    Click
  </button>
</div>
`;
